build.gradle

```diff
@@ -37,20 +37,29 @@ allprojects {
 
         // Use Scala 2.12 in our library project
         implementation 'org.scala-lang:scala-library:2.12.7'
-        implementation 'com.typesafe.scala-logging:scala-logging_2.12:3.9.2'
+        implementation "com.typesafe.scala-logging:scala-logging_$scalaVersion:3.9.2"
 
         // Use Scalatest for testing our library
         testImplementation 'junit:junit:4.12'
-        testImplementation "org.scalatest:scalatest_$scalaVersion:3.0.5"
+        compile "org.scalatest:scalatest_$scalaVersion:3.0.5"
 
         // Need scala-xml at test runtime
         testRuntimeOnly "org.scala-lang.modules:scala-xml_$scalaVersion:1.1.1"
 
         // Logging
         compile group: 'com.typesafe.scala-logging', name: "scala-logging_$scalaVersion", version: '3.9.2'
+        testRuntime group: 'ch.qos.logback', name: 'logback-classic', version: '1.1.7'
 
         // Akka
         compile group: 'com.typesafe.akka', name: "akka-http_$scalaVersion", version: '10.0.11'
+
+        // Apache Curator. See this [compatibility documentation](http://curator.apache.org/zk-compatibility.html)
+        def withoutZk = { exclude group: 'org.apache.zookeeper', module: 'zookeeper' }
```

Comment:
should really all projects depend on curator? Even metrics? I think we should move this to `test-utils/build.gradle` right now

Reply:
I did that initially but then I realized that `test-utils`, `persistence-zookeeper`, frameworks and future leader election code will depend on it so I decided to move it up. wdyt?